### ISSUE TITLE
Crash when setting custom keyboard controls

### ISSUE THREAD

**Issue by ibara** [omf2097/openomf#351]
Hi --

When attempting to set custom keyboard controls, the game locks up. You can enter the custom keyboard menu, select a key to change but once you press the new key to map to, the game locks up. Unfortunately, I could not get anything useful from the game itself to help narrow this down.

This is on OpenBSD/amd64 (and to note from #59, OpenBSD requires no patches to play).

Other than that, OpenOMF works great and is on its way to having an official OpenBSD package.

Thanks!

**Comment by Vagabond** [comment]
It seems to have to do with a disabled component hanging in `menu_action`.

I'm not sure how to fix it properly, but this patch seems to help:

```patch
diff --git a/src/game/gui/menu.c b/src/game/gui/menu.c
index d4aa430..9679ab1 100644
--- a/src/game/gui/menu.c
+++ b/src/game/gui/menu.c
@@ -157,7 +157,7 @@ static int menu_action(component *mc, int action) {
                 // Update selected component
                 c = sizer_get(mc, m->selected);

-            } while(component_is_disabled(c));
+            } while(component_is_disabled(c) && c->supports_disable);
             // Play menu sound
             sound_play(19, 0.5f, 0.0f, 2.0f);
             component_select(c, 1);
```

**Comment by ibara** [comment]
That does the trick for me.

**Comment by katajakasa** [comment]
Now fixed in master @ 73b63423c03690395c03a4551801a40b7edd4849
